container/shim/src/config.js

```diff
@@ -66,12 +67,14 @@ function pVersion(version) {
 }
 
 async function readOrCreateNodeId() {
+  let nodeId;
   try {
-    const nodeId = await readFile(NODE_ID_FILE_PATH, "utf8");
-    return nodeId.trim();
+    nodeId = (await readFile(NODE_ID_FILE_PATH, "utf8")).trim();
   } catch (err) {
-    return createNodeId();
+    nodeId = createNodeId();
   }
+  writeFile(NODE_ID_NGINX_CONF, `set $node_id "${nodeId}";`).catch(console.error); // eslint-disable-line no-console
```

Comment:
the first time this file is created, nginx will be already running so this file will not be included and nginx will need to restart or reload config right ?

Reply:
correct, that's why the `if` was there :/